content/sensu-core/1.0/guides/pre-compile-plugins.md

```diff
@@ -0,0 +1,146 @@
+
+We'll use `gem-compiler` to rebuild the gem packages with the included native extensions:
+
+```bash
+cd /tmp/gems/cache
+gem compile unf_ext-0.0.7.4.gem
+gem compile nokogiri-1.8.1.gem --prune
+```
+
+_NOTE: `nokogiri` requires --prune flag per gem-compiler readme notes _
+
+The resulting compiled gem files have a x86_64-linux in their file name denoting their system architech:
+
+```bash
+$ ls  -1 *-x86_64-linux.gem
+nokogiri-1.8.1-x86_64-linux.gem
+unf_ext-0.0.7.4-x86_64-linux.gem
+```
+
+## Install Pre-Compiled Gems on Targeted System and Testing
+
+### Install compiled gem into embedded Ruby
+Now we can copy these gems to another system, install them into the sensu embedded ruby. 
+In a production or development enviroment, you'd want to put these in a rubygems server of your own (e.g. [Geminabox(https://github.com/geminabox/geminabox), [Artifactory](https://jfrog.com/artifactory/) or similar). You can also use a tool like `fpm` to convert the gems into a system package.
+
+Here we've copied the compiled gems to a system with Sensu installed, but no gcc or other compile toolchain
+
+ _NOTE: nokogiri needs libxml2 and possibly libxslt to be present on your system._
+
+
+```bash
+$ /opt/sensu/embedded/bin/gem install unf_ext-0.0.7.4-x86_64-linux.gem
+$ /opt/sensu/embedded/bin/gem install nokogiri-1.8.1-x86_64-linux.gem
+```
+
+### Install Sensu plugins as normal
+With these prerequisites in place we can install sensu-plugins-aws without a compiler. You can use `sensu-install` or `gem` commands:
+
+```bash
+$ sensu-install -p aws
+[SENSU-INSTALL] installing Sensu plugins ...
+[SENSU-INSTALL] determining if Sensu gem 'sensu-plugins-aws' is already installed ...
+false
+[SENSU-INSTALL] Sensu plugin gems to be installed: ["sensu-plugins-aws"]
+[SENSU-INSTALL] installing Sensu gem 'sensu-plugins-aws'
+Fetching: unf-0.1.4.gem (100%)
+Successfully installed unf-0.1.4
+Fetching: domain_name-0.5.20170404.gem (100%)
+Successfully installed domain_name-0.5.20170404
+Fetching: http-cookie-1.0.3.gem (100%)
+Successfully installed http-cookie-1.0.3
+Fetching: mime-types-2.99.3.gem (100%)
+Successfully installed mime-types-2.99.3
+Fetching: netrc-0.11.0.gem (100%)
+Successfully installed netrc-0.11.0
+Fetching: rest-client-1.8.0.gem (100%)
+Successfully installed rest-client-1.8.0
+Fetching: sensu-plugins-aws-10.0.3.gem (100%)
+You can use the embedded Ruby by setting EMBEDDED_RUBY=true in /etc/default/sensu
+Successfully installed sensu-plugins-aws-10.0.3
+7 gems installed
+[SENSU-INSTALL] successfully installed Sensu plugins: ["sensu-plugins-aws"]  
+```
+
+To test our pre-compiled gems we can use `check-elb-health-fog.rb` as it should exercise the nokogiri dependency:
+
+```bash
+$ export AWS_ACCESS_KEY=fatchance
+$ export AWS_SECRET_KEY=noway
+$ /opt/sensu/embedded/bin/check-elb-health-fog.rb -n foo -r us-east-1
+ELBHealth WARNING: An issue occured while communicating with the AWS EC2 API: 
+There is no ACTIVE Load Balancer named 'foo' 
+```
```

Comment:
This is a bit of an odd ending. It just drops off without any call to action. I'd recommend an invitation to edit the doc, or submit an issue. You'll probably want to recap what you went over too.

Reply:
updated with a nicer ending